zerver/lib/zulip_update_announcements.py

```diff
@@ -0,0 +1,206 @@
+from django.utils.timezone import now as timezone_now
+from django.utils.translation import override as override_language
+
+from zerver.actions.message_send import (
+    do_send_messages,
+    internal_prep_huddle_message,
+    internal_prep_stream_message,
+)
+from zerver.lib.message import SendMessageRequest
+from zerver.models.realm_audit_logs import RealmAuditLog
+from zerver.models.realms import Realm
+from zerver.models.users import UserProfile, get_system_bot
+
+
+@dataclass
+class ZulipUpdateAnnouncement:
+    level: int
+    message: str
+
+
+# We don't translate the announcement message because they are quite unlikely to be
+# translated during the time between when we draft them and when they are published.
+zulip_update_announcements: List[ZulipUpdateAnnouncement] = [
+    ZulipUpdateAnnouncement(
+        level=1,
+        message=(
+            "This topic will receive automated updates about new Zulip features.\n"
+            "\n"
+            "If you like, you can [configure]({settings_url}) a different stream "
+            "for these updates (and [move]({move_content_help_url}) the initial updates there), "
+            "or [turn this feature off]({settings_url}) altogether."
+        ).format(
+            settings_url="/#organization/organization-settings",
+            move_content_help_url="/help/move-content-to-another-stream",
+        ),
+    ),
+]
+
+
+def get_latest_zulip_update_announcements_level() -> int:
+    latest_zulip_update_announcement = zulip_update_announcements[-1]
+    return latest_zulip_update_announcement.level
+
+
+def get_zulip_update_announcements_message_for_level(level: int) -> str:
+    zulip_update_announcement = zulip_update_announcements[level - 1]
+    return zulip_update_announcement.message
+
+
+def get_realms_behind_zulip_update_announcements_level(level: int) -> QuerySet[Realm]:
+    # Filter out deactivated realms. When a realm is later
+    # reactivated, send the notices it missed while it was deactivated.
+    realms = Realm.objects.filter(
+        Q(zulip_update_announcements_level__isnull=True)
+        | Q(zulip_update_announcements_level__lt=level),
+        deactivated=False,
+    ).exclude(string_id=settings.SYSTEM_BOT_REALM)
+    return realms
+
+
+def internal_prep_group_direct_message_for_old_realm(
+    realm: Realm, sender: UserProfile
+) -> Optional[SendMessageRequest]:
+    administrators = list(realm.get_human_admin_users())
+    with override_language(realm.default_language):
+        topic_name = str(realm.ZULIP_UPDATE_ANNOUNCEMENTS_TOPIC_NAME)
+    if realm.zulip_update_announcements_stream is None:
+        content = (
+            "You can now [configure]({organization_settings_url}) a stream where Zulip "
+            "will send updates to users in your organization about new Zulip features. "
+            "These notifications are currently turned off in your organization."
+        ).format(organization_settings_url="/#organization/organization-settings")
```

Comment:
This is a bit had to read. Let's try "will send updates to users in your organization" -> "will send updates".